CrazyCanvas/Source/ECS/Systems/Player/HealthSystem.cpp

```diff
@@ -0,0 +1,121 @@
+#include "ECS/Systems/Player/HealthSystem.h"
+#include "ECS/ECSCore.h"
+#include "ECS/Components/Player/HealthComponent.h"
+
+#include "Application/API/Events/EventQueue.h"
+
+#include <mutex>
+
+HealthSystem::HealthSystem()
+	: m_HealthEntities()
+{
+}
+
+HealthSystem::~HealthSystem()
+{
+	using namespace LambdaEngine;
+	EventQueue::UnregisterEventHandler<ProjectileHitEvent>(this, &HealthSystem::OnProjectileHit);
+}
+
+bool HealthSystem::OnProjectileHit(const ProjectileHitEvent& projectileHitEvent)
+{
+	using namespace LambdaEngine;
+
+	std::scoped_lock<SpinLock> lock(m_DefferedEventsLock);
+	LOG_INFO("Something got hit CollisionInfo0=%d, CollisionInfo1=%d", projectileHitEvent.CollisionInfo0.Entity, projectileHitEvent.CollisionInfo1.Entity);
+	
+	for (Entity entity : m_HealthEntities)
+	{
+		LOG_INFO("....Entity: %d", entity);
+
+		// CollisionInfo0 is the projectile
+		if (projectileHitEvent.CollisionInfo1.Entity == entity)
+		{
+			m_DeferredHitEvents.EmplaceBack(projectileHitEvent);
+			LOG_INFO("Player got hit");
+			break;
+		}
+	}
+
+	return true;
+}
+
+bool HealthSystem::Init()
+{
+	using namespace LambdaEngine;
+
+	// Register system
+	{
+		SystemRegistration systemReg = {};
+		systemReg.SubscriberRegistration.EntitySubscriptionRegistrations =
+		{
+			{
+				.pSubscriber = &m_HealthEntities,
+				.ComponentAccesses =
+				{
+					{ RW, HealthComponent::Type() }
+				}
+			}
+		};
+
+		// After weaponsystem -> Do not know if this is correct
+		systemReg.Phase = 2;
+
+		RegisterSystem(systemReg);
+	}
+
+	// Register eventhandler
+	EventQueue::RegisterEventHandler<ProjectileHitEvent>(this, &HealthSystem::OnProjectileHit);
+
+	return true;
+}
+
+void HealthSystem::Tick(LambdaEngine::Timestamp deltaTime)
+{
+	using namespace LambdaEngine;
+
+	ECSCore* pEcs = ECSCore::GetInstance();
+	ComponentArray<HealthComponent>* healthComponents = pEcs->GetComponentArray<HealthComponent>();
+
+	// Since the events are not sent threadsafe
+	{
+		std::scoped_lock<SpinLock> lock(m_DefferedEventsLock);
+		if (!m_DeferredHitEvents.IsEmpty())
+		{
+			m_EventsToProcess = m_DeferredHitEvents;
+			m_DeferredHitEvents.Clear();
+		}
+	}
+
+	if (!m_EventsToProcess.IsEmpty())
+	{
+		for (ProjectileHitEvent& event : m_EventsToProcess)
```

Comment:
Unnecessary if-statement

Reply:
Having this statement results in only one check instead of two. Without it, there is one in the loop and the other one in Clear.